### PR TITLE
Multi text input + Goals input

### PR DESCRIPTION
**Pull request by IvanYu327**
## Notion ticket link

[Create multi-text field component](https://www.notion.so/uwblueprintexecs/Create-multi-text-field-component-9bf15e7f71bc47a2b8423c188c0867cc)

## Implementation description
* Created a div that is an array of divs with an input field at the end to simulate a single input field with tags
* Created autocomplete to select tags
* Ability to add and delete tags with enter and delete keypresses

## Steps to test
1. go to `http://localhost:3000/intake`
2. go to the Program details
3. Enter Familial concerns
4. Entering text should behave like a text input
5. Pressing enter with text should create a tag
6. Pressing enter without text should do nothing
7. Pressing backspace while there's no text should delete the last tag
8. Pressing backspace while there is text should backspace text as usual
9. Presssing the X on a tag should delete it
10. Clicking on the text input should open the autocomplete
11. Clicking autocomplete option should add that tag to the field
12. Entering text that is not on the suggestions should show no autocomplete

## What should reviewers focus on?
* `MultiTextInput.tsx`

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/common/MultiTextInput.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
+import React, { useState, useRef, useEffect } from "react";
+import { Box, Flex, Input } from "@chakra-ui/react";
+import { X } from "react-feather";
+
+type MultiTextInputProps = {
+  placeholder: string;
+  options: string[];
+  values: string[];
+  setState: (e: string[]) => void;
```
Comment (bowenzhu1) [inline frontend/src/components/common/MultiTextInput.tsx:9]: let's rename this to be a bit more descriptive, maybe `(newValue: string[]) => void` or something

File: frontend/src/components/common/MultiTextInput.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+      return option.toLowerCase().startsWith(inputValue.toLowerCase());
+    });
+
+    setFilteredOptions(newFilteredOptions);
+    setShowAutocomplete(newFilteredOptions.length !== 0);
+  }, [inputValue, options]);
+
+  useEffect(() => {
+    setShowAutocomplete(false);
+  }, []);
```
Comment (bowenzhu1) [inline frontend/src/components/common/MultiTextInput.tsx:53]: do we need this? I though we're already setting showAutoComplete to false when initializing the state

File: frontend/src/components/common/MultiTextInput.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        setShowAutocomplete(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleMouseDown);
+
+    return () => {
+      document.removeEventListener("mousedown", handleMouseDown);
+    };
+  }, []);
```
Comment (bowenzhu1) [inline frontend/src/components/common/MultiTextInput.tsx:40]: some bugs with displaying the autocomplete as mentioned via discord (it shows it on initial render + when clicking on other random form fields)

File: frontend/src/components/common/MultiTextInput.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        borderBottomRadius={showAutocomplete ? "0px" : "4px"}
+        borderColor="gray.100"
+        backgroundColor="gray.50"
+        _hover={{ backgroundColor: "gray.100" }}
+      >
+        <Flex flexWrap="wrap">
+          {values &&
+            values.map((tag, i) => {
+              return (
+                <Box
```
Comment (bowenzhu1) [inline frontend/src/components/common/MultiTextInput.tsx:92]: change style to match the style of existing selector tags

File: frontend/src/components/common/MultiTextInput.tsx [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+          >
+            <Input
+              ref={inputRef}
+              variant="unstyled"
+              borderRadius={0}
+              placeholder={placeholder}
+              value={inputValue}
+              onChange={(e) => setInputValue(e.target.value)}
+              onKeyDown={handleKeyPress}
+              onFocus={() => setShowAutocomplete(true)}
```
Comment (bowenzhu1) [inline frontend/src/components/common/MultiTextInput.tsx:132]: this onFocus doesn't seem to be performing as expected

File: frontend/src/components/common/MultiTextInput.tsx [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+            />
+          </Box>
+        </Flex>
+      </Box>
+      <Box
+        ref={autocompleteRef}
+        display={showAutocomplete && filteredOptions.length > 0 ? "" : "none"}
```
Comment (bowenzhu1) [inline frontend/src/components/common/MultiTextInput.tsx:139]: can we also add a way to disable this via a prop like `readOnly` - so there's no way to input or see the list

**Review by bowenzhu1 — CHANGES_REQUESTED** [review]
great work! left some comments + follow up notes from our discord talk

**Comment by IvanYu327** [comment]
Added the short/long term goals things here too. Testing them is the same, just that there are now 3 MultiTextInput components on the Program Details form to check instead of 1.

**Review by bowenzhu1 — APPROVED** [review]
looks good now! awesome work 

there's one small minor behavior on my end where I can select the end of the component textbox and it seems like i'm able to type. Not an issue really, but feel free to spend ~5 minutes poking around at it. picture below for reference:
image

**Pull request merged**